pkg/plugins/policies/meshmetric/plugin/v1alpha1/plugin.go

```diff
@@ -257,7 +265,18 @@ func createDynamicConfig(conf api.Conf, proxy *core_xds.Proxy, resources xds_con
 	if rawList := resources[core_mesh.MeshGatewayType]; rawList != nil {
 		gateways = rawList.(*core_mesh.MeshGatewayResourceList).Items
 	}
-	extraLabels := mads.DataplaneLabels(proxy.Dataplane, gateways)
+
+	var extraLabels map[string]string
```

Comment:
To make my later comment unecessary because it'd be obvious I'd do:

```
extraLabels = map[string]string{
   WorkloadAttributeKey : "TODO"
}
if !unified_naming.Enabled(proxy.Metadata, mesh) {
   // add legacy label
}
```

This also makes cleaning the backward compatibility easier (just delete the if block)